pkg/kong/globalrules_test.go

```diff
@@ -0,0 +1,169 @@
+package kong
+
+import (
+	"testing"
+
+	"github.com/api7/kong-to-apisix/pkg/apisix"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMigrateGlobalRules(t *testing.T) {
+	var err error
+	var kongConfig Config
+	var apisixConfig apisix.Config
+	var kongPluginKeyAuth Plugin
+	kongPluginKeyAuth.ID = uuid.NewV4().String()
+	kongPluginKeyAuth.Name = PluginKeyAuth
+	kongPluginKeyAuth.Enabled = true
+	kongPluginKeyAuthConfig := make(map[string]interface{})
+	kongPluginKeyAuthConfig["key_names"] = make([]interface{}, 1)
+	kongPluginKeyAuthConfigKeyNames := make([]interface{}, 1)
+	for i, v := range []string{"apikey"} {
+		kongPluginKeyAuthConfigKeyNames[i] = v
+	}
+	kongPluginKeyAuthConfig["key_names"] = kongPluginKeyAuthConfigKeyNames
+	kongPluginKeyAuth.Config = kongPluginKeyAuthConfig
+	kongConfig.Plugins = append(kongConfig.Plugins, kongPluginKeyAuth)
+	err = MigrateGlobalRules(&kongConfig, &apisixConfig)
+	assert.NoError(t, err)
+	assert.Equal(t, kongPluginKeyAuth.ID, apisixConfig.GlobalRules[0].ID)
+
+	kongConfig = Config{}
+	apisixConfig = apisix.Config{}
+	var kongPluginRateLimitingSecond Plugin
+	kongPluginRateLimitingSecond.ID = uuid.NewV4().String()
+	kongPluginRateLimitingSecond.Name = PluginRateLimiting
+	kongPluginRateLimitingSecond.Enabled = true
+	kongPluginRateLimitingSecondConfig := make(map[string]interface{})
+	kongPluginRateLimitingSecondConfig["second"] = 5
```

Comment:
Better to use table drive test: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
done